src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -206,7 +206,10 @@ private void WriteMethodBody(IMethodDefinition method)
                 Write("throw new ");
                 if (_forCompilationIncludeGlobalprefix)
                     Write("global::");
-                Write("System.PlatformNotSupportedException(); ");
+                if(_platformNotSupportedExceptionMessage == null)
+                    Write("System.PlatformNotSupportedException();");
+                else
+                    Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
```

Comment:
Can you also provide some syntax pattern for folks specify a resource string?

For example, if _platformNotSupportedExceptionMessage starts with some character like `:` or `@` treat the rest of it as code so that folks can do something like `:SR.MyLocalizedResourceMessage`